src/MyParcelComApi.php

```diff
@@ -837,17 +839,50 @@ private function arrayToFilter(array $array)
     {
         $filters = [];
         foreach ($array as $name => $value) {
-            /**
-             * @deprecated This condition is to keep the SDK backwards
-             * compatible until the API has been updated to use the new filter.
-             */
-            if ($name === 'has_active_contract') {
-                $filters['filter[has_contract]'] = $value;
-            }
-
             $filters["filter[$name]"] = $value;
         }
 
         return $filters;
     }
+
+    /**
+     * Determines which carriers to look pudo locations up for.
+     * The specificCarrier parameter indicates a specific carrier to look up pudo locations for. Otherwise,
+     * all carriers will be used.
+     * The onlyActiveContracts parameter indicates whether only carriers for which the user has an active contract
+     * for services with delivery method pickup should be used for pudo location retrieval.
+     *
+     * @param bool                  $onlyActiveContracts
+     * @param null|CarrierInterface $specificCarrier
+     * @return array
+     */
+    private function determineCarriersForPudoLocations($onlyActiveContracts, $specificCarrier = null)
+    {
+        // If we're looking for a specific carrier but it doesn't
+        // matter if it has active contracts, just return it immediately.
+        if (!$onlyActiveContracts && $specificCarrier) {
+            return [$specificCarrier];
+        }
+
+        // Return all carriers if we're not filtering for anything
+        // specific.
+        if (!$onlyActiveContracts && !$specificCarrier) {
```

Comment:
You can reduce this check to just `if (!$onlyActiveContracts)` as the previous check already ensures that `$specificCarrier` will be falsy if `$onlyActiveContract` is falsy.